src/main/frontend/Admin.html

```diff
@@ -197,88 +238,101 @@ <h1>Flights Database</h1>
+
+        // Iterate over the fields and create textboxes for each
         fields.forEach(field => {
-            updatedData[field] = form.elements[field].value;
-        });
-        console.log(JSON.stringify(updatedData))
-        // Perform the fetch request
-        fetch('http://127.0.0.1:5000/scene/update', {
-            method: 'POST',
-            headers: {
-                'Content-Type': 'application/json',
-            },
-            body: JSON.stringify(updatedData),
-        })
-        .then(response => {
-            if (!response.ok) {
-                throw new Error(`HTTP error! Status: ${response.status}`);
+            let label = document.createElement("label");
+            label.textContent = `Enter the new ${field}: `;
+            let input = document.createElement("input");
+
+            // Adjust the cell index for RunWay and GATE
+            let cellIndex;
+            if (field === "RunWay") {
+                cellIndex = 12;
+            } else if (field === "GATE") {
+                cellIndex = 13;
+            } else if (field === "STACK") {
+                cellIndex = 14;
+            } else {
+                cellIndex = fields.indexOf(field) + 2;
             }
-            return response.json();
-        })
-        .then(data => {
-            console.log(data.message);
-            // Optionally update your UI based on the response
-        })
-        .catch(error => {
-            console.error('Error updating data:', error.message);
+
+            input.value = row.cells[cellIndex].innerHTML;
+            input.type = "text";
+            input.name = field;
+            form.appendChild(label);
+            form.appendChild(input);
+            form.appendChild(document.createElement("br"));
         });
 
-        // Remove the modal overlay after submission
-        document.body.removeChild(modalOverlay);
-    });
-}
+        // Add a submit button to the form
+        let submitButton = document.createElement("button");
+        submitButton.textContent = "Submit";
+        form.appendChild(submitButton);
+
+        // Create a modal overlay
+        let modalOverlay = document.createElement("div");
+        modalOverlay.classList.add("modal-overlay");
+
+        // Display the form in the modal overlay
+        modalOverlay.appendChild(form);
+        document.body.appendChild(modalOverlay);
+
+        // Handle the form submission
+        form.addEventListener("submit", function (event) {
+            event.preventDefault();
+
+            // Extract values from the form
+            let updatedData = { FLIGHT_ID: row.cells[1].innerHTML };
+            fields.forEach(field => {
+                updatedData[field] = form.elements[field].value;
+            });
+            console.log(JSON.stringify(updatedData))
+            // Perform the fetch request
+            fetch('http://127.0.0.1:5000/scene/update', {
```

Comment:
Look into seting up a non hardcoded url. Something like an env file would be better.
Its really easy if you make a env.js file and then just export all the values you want